
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="..">
      
      
        <link rel="next" href="../examples/">
      
      
      <link rel="icon" href="../static/talus-logo.svg">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.12">
    
    
      
        <title>API - gopher</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.2afb09e1.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=IBM+Plex+Sans:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"IBM Plex Sans";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../assets/_mkdocstrings.css">
    
      <link rel="stylesheet" href="../stylesheets/extra.css">
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    <body dir="ltr">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#api" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="gopher" class="md-header__button md-logo" aria-label="gopher" data-md-component="logo">
      
  <img src="../static/talus-logo.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            gopher
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              API
            
          </span>
        </div>
      </div>
    </div>
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/TalusBio/gopher" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href=".." class="md-tabs__link">
        
  
  
    
  
  Home

      </a>
    </li>
  

      
        
  
  
  
    
  
  
    <li class="md-tabs__item md-tabs__item--active">
      <a href="./" class="md-tabs__link">
        
  
  
    
  
  API

      </a>
    </li>
  

      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../examples/" class="md-tabs__link">
        
  
  
    
  
  Examples

      </a>
    </li>
  

      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../vignettes/" class="md-tabs__link">
        
  
  
    
  
  Enrichment vignette

      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


  

<nav class="md-nav md-nav--primary md-nav--lifted md-nav--integrated" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="gopher" class="md-nav__button md-logo" aria-label="gopher" data-md-component="logo">
      
  <img src="../static/talus-logo.svg" alt="logo">

    </a>
    gopher
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/TalusBio/gopher" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    API
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    API
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#gopher.read_encyclopedia" class="md-nav__link">
    <span class="md-ellipsis">
      read_encyclopedia
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#gopher.read_metamorpheus" class="md-nav__link">
    <span class="md-ellipsis">
      read_metamorpheus
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#gopher.test_enrichment" class="md-nav__link">
    <span class="md-ellipsis">
      test_enrichment
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#gopher.get_data_dir" class="md-nav__link">
    <span class="md-ellipsis">
      get_data_dir
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#gopher.set_data_dir" class="md-nav__link">
    <span class="md-ellipsis">
      set_data_dir
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#gopher.generate_annotations" class="md-nav__link">
    <span class="md-ellipsis">
      generate_annotations
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#gopher.load_annotations" class="md-nav__link">
    <span class="md-ellipsis">
      load_annotations
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#gopher.get_annotations" class="md-nav__link">
    <span class="md-ellipsis">
      get_annotations
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#gopher.map_proteins" class="md-nav__link">
    <span class="md-ellipsis">
      map_proteins
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#gopher.normalize_values" class="md-nav__link">
    <span class="md-ellipsis">
      normalize_values
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#gopher.get_rankings" class="md-nav__link">
    <span class="md-ellipsis">
      get_rankings
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#gopher.in_term" class="md-nav__link">
    <span class="md-ellipsis">
      in_term
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#gopher.roc" class="md-nav__link">
    <span class="md-ellipsis">
      roc
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../examples/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Examples
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../vignettes/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Enrichment vignette
    
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="api">API</h1>


<div class="doc doc-object doc-function">


<h3 id="gopher.read_encyclopedia" class="doc doc-heading">
            <code class="highlight language-python"><span class="n">gopher</span><span class="o">.</span><span class="n">read_encyclopedia</span><span class="p">(</span><span class="n">proteins_txt</span><span class="p">)</span></code>

</h3>


    <div class="doc doc-contents first">

        <p>Read results from EncyclopeDIA.</p>


<table>
      <thead>
        <tr>
          <th><span class="doc-section-title">PARAMETER</span></th>
          <th><span>DESCRIPTION</span></th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>proteins_txt</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>The EncyclopeDIA protein output.</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="str">str</span></code>
                  </span>
              </p>
            </td>
          </tr>
      </tbody>
    </table>


<table>
      <thead>
        <tr>
          <th><span class="doc-section-title">RETURNS</span></th>
          <th><span>DESCRIPTION</span></th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <span class="doc-returns-annotation">
                    <code><span title="pandas.DataFrame">DataFrame</span></code>
                </span>
            </td>
            <td class="doc-returns-details">
              <div class="doc-md-description">
                <p>The EncyclopeDIA results in a format for gopher.</p>
              </div>
            </td>
          </tr>
      </tbody>
    </table>


    </div>

</div>

<div class="doc doc-object doc-function">


<h3 id="gopher.read_metamorpheus" class="doc doc-heading">
            <code class="highlight language-python"><span class="n">gopher</span><span class="o">.</span><span class="n">read_metamorpheus</span><span class="p">(</span><span class="n">proteins_txt</span><span class="p">)</span></code>

</h3>


    <div class="doc doc-contents first">

        <p>Read results from Metamorpheus.</p>


<table>
      <thead>
        <tr>
          <th><span class="doc-section-title">PARAMETER</span></th>
          <th><span>DESCRIPTION</span></th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>proteins_txt</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>The Metamorpheus protein output file.</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="str">str</span></code>
                  </span>
              </p>
            </td>
          </tr>
      </tbody>
    </table>


<table>
      <thead>
        <tr>
          <th><span class="doc-section-title">RETURNS</span></th>
          <th><span>DESCRIPTION</span></th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <span class="doc-returns-annotation">
                    <code><span title="pandas.DataFrame">DataFrame</span></code>
                </span>
            </td>
            <td class="doc-returns-details">
              <div class="doc-md-description">
                <p>The Metamorpheus results in a format for gopher.</p>
              </div>
            </td>
          </tr>
      </tbody>
    </table>


    </div>

</div>

<div class="doc doc-object doc-function">


<h3 id="gopher.test_enrichment" class="doc doc-heading">
            <code class="highlight language-python"><span class="n">gopher</span><span class="o">.</span><span class="n">test_enrichment</span><span class="p">(</span><span class="n">proteins</span><span class="p">,</span> <span class="n">desc</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">aspect</span><span class="o">=</span><span class="s1">&#39;all&#39;</span><span class="p">,</span> <span class="n">species</span><span class="o">=</span><span class="s1">&#39;human&#39;</span><span class="p">,</span> <span class="n">release</span><span class="o">=</span><span class="s1">&#39;current&#39;</span><span class="p">,</span> <span class="n">go_subset</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">contaminants_filter</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">fetch</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">progress</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">annotations</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">mapping</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">aggregate_terms</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">alternative</span><span class="o">=</span><span class="s1">&#39;greater&#39;</span><span class="p">)</span></code>

</h3>


    <div class="doc doc-contents first">

        <p>Test for the enrichment of Gene Ontology terms from protein abundance.</p>
<p>The Mann-Whitney U Test is applied to each column of proteins dataframe and
for each Gene Ontology (GO) term. The p-values are then corrected for
multiple hypothesis testing across all of the columns using the
Benjamini-Hochberg procedure.</p>


<table>
      <thead>
        <tr>
          <th><span class="doc-section-title">PARAMETER</span></th>
          <th><span>DESCRIPTION</span></th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>proteins</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>A dataframe where the indices are UniProt accessions and each column is
an experiment to test. The values in this dataframe should be some
measure of protein abundance: these could be the raw measurement if
originating from a single sample or a fold-change/p-value if looking at
the difference between two conditions.</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="pandas.DataFrame">DataFrame</span></code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>desc</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>Rank proteins in descending order?</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="bool">bool</span></code>
                  </span>
                  <span class="doc-param-default">
                    <b>DEFAULT:</b>
                      <code>True</code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>aspect</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>The Gene Ontology aspect to use. Use &ldquo;cc&rdquo; for &ldquo;Cellular Compartment&rdquo;,
&ldquo;mf&rdquo; for &ldquo;Molecular Function&rdquo;, &ldquo;bp&rdquo; for &ldquo;Biological Process&rdquo;, or &ldquo;all&rdquo;
for all three.</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code>(<span title="str">str</span>, {<span title="cc">cc</span>, <span title="mf">mf</span>, <span title="bp">bp</span>, <span title="all">all</span>})</code>
                  </span>
                  <span class="doc-param-default">
                    <b>DEFAULT:</b>
                      <code>&#39;all&#39;</code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>species</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>The species for which to retrieve GO annotations. If not &ldquo;human&rdquo; or
&ldquo;yeast&rdquo;, see
<a href="http://current.geneontology.org/products/pages/downloads.html">here</a>.</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code>str, {&#34;human&#34;, &#34;yeast&#34;, ...}, optional.</code>
                  </span>
                  <span class="doc-param-default">
                    <b>DEFAULT:</b>
                      <code>&#39;human&#39;</code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>release</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>The Gene Ontology release version. Using &ldquo;current&rdquo; will look up the
most current version.</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="str">str</span></code>
                  </span>
                  <span class="doc-param-default">
                    <b>DEFAULT:</b>
                      <code>&#39;current&#39;</code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>go_subset</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>The go terms of interest. Should consists of the go term names such
as &lsquo;nucleus&rsquo; or &lsquo;cytoplasm&rsquo;.</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="list">list</span></code>
                  </span>
                  <span class="doc-param-default">
                    <b>DEFAULT:</b>
                      <code>None</code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>contaminants_filter</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>A list of uniprot accessions for common contaminants such as
Keratin to filter out.</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="list">list</span></code>
                  </span>
                  <span class="doc-param-default">
                    <b>DEFAULT:</b>
                      <code>None</code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>fetch</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>Download the GO annotations even if they have been downloaded before?</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="bool">bool</span></code>
                  </span>
                  <span class="doc-param-default">
                    <b>DEFAULT:</b>
                      <code>False</code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>progress</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>Show a progress bar during enrichment tests?</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="bool">bool</span></code>
                  </span>
                  <span class="doc-param-default">
                    <b>DEFAULT:</b>
                      <code>False</code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>annotations</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>A custom annotations dataframe.</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="pandas.DataFrame">DataFrame</span></code>
                  </span>
                  <span class="doc-param-default">
                    <b>DEFAULT:</b>
                      <code>None</code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>mapping</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>A custom mapping of the GO term relationships.</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="dict">dict</span></code>
                  </span>
                  <span class="doc-param-default">
                    <b>DEFAULT:</b>
                      <code>None</code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>aggregate_terms</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>Aggregate the terms and do the graph search.</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="bool">bool</span></code>
                  </span>
                  <span class="doc-param-default">
                    <b>DEFAULT:</b>
                      <code>True</code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>alternative</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>Type of test that should be run.
Could be &ldquo;greater&rdquo;, &ldquo;less&rdquo;, or &ldquo;two-sided&rdquo;.</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code>str, {&#34;greater&#34;, &#34;less&#34;, &#34;two-sided&#34;} optional</code>
                  </span>
                  <span class="doc-param-default">
                    <b>DEFAULT:</b>
                      <code>&#39;greater&#39;</code>
                  </span>
              </p>
            </td>
          </tr>
      </tbody>
    </table>


<table>
      <thead>
        <tr>
          <th><span class="doc-section-title">RETURNS</span></th>
          <th><span>DESCRIPTION</span></th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <span class="doc-returns-annotation">
                    <code><span title="pandas.DataFrame">DataFrame</span></code>
                </span>
            </td>
            <td class="doc-returns-details">
              <div class="doc-md-description">
                <p>The adjusted p-value for each tested GO term in each sample.</p>
              </div>
            </td>
          </tr>
      </tbody>
    </table>


    </div>

</div>

<div class="doc doc-object doc-function">


<h3 id="gopher.get_data_dir" class="doc doc-heading">
            <code class="highlight language-python"><span class="n">gopher</span><span class="o">.</span><span class="n">get_data_dir</span><span class="p">()</span></code>

</h3>


    <div class="doc doc-contents first">

        <p>Retrieve the current data directory for ppx.</p>


    </div>

</div>

<div class="doc doc-object doc-function">


<h3 id="gopher.set_data_dir" class="doc doc-heading">
            <code class="highlight language-python"><span class="n">gopher</span><span class="o">.</span><span class="n">set_data_dir</span><span class="p">(</span><span class="n">path</span><span class="o">=</span><span class="kc">None</span><span class="p">)</span></code>

</h3>


    <div class="doc doc-contents first">

        <p>Set the ppx data directory.</p>


<table>
      <thead>
        <tr>
          <th><span class="doc-section-title">PARAMETER</span></th>
          <th><span>DESCRIPTION</span></th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>The path for ppx to use as its data directory.</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code>str or pathlib.Path object</code>
                  </span>
                  <span class="doc-param-default">
                    <b>DEFAULT:</b>
                      <code>None</code>
                  </span>
              </p>
            </td>
          </tr>
      </tbody>
    </table>


    </div>

</div>

<div class="doc doc-object doc-function">


<h3 id="gopher.generate_annotations" class="doc doc-heading">
            <code class="highlight language-python"><span class="n">gopher</span><span class="o">.</span><span class="n">generate_annotations</span><span class="p">(</span><span class="n">proteins</span><span class="p">,</span> <span class="n">aspect</span><span class="p">,</span> <span class="n">go_name</span><span class="p">,</span> <span class="n">go_id</span><span class="o">=</span><span class="kc">None</span><span class="p">)</span></code>

</h3>


    <div class="doc doc-contents first">

        <p>Generate an annotation file for a list of proteins that are correlated to a single term and aspect.</p>
<p>The term can be in the GO database or a new term.</p>


<table>
      <thead>
        <tr>
          <th><span class="doc-section-title">PARAMETER</span></th>
          <th><span>DESCRIPTION</span></th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>proteins</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>List of proteins (UniProtKB accessions) that will be annotated to a term.</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="List">List</span>[<span title="str">str</span>]</code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>aspect</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>String specifying the aspect the term is in (&ldquo;C&rdquo;, &ldquo;F&rdquo;, &ldquo;P&rdquo;).</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="str">str</span></code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>go_name</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>String of the GO name for the proteins</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="str">str</span></code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>go_id</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>String of the GO ID. If in the GO database, the go id and go name should match the database.</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="str">str</span></code>
                  </span>
                  <span class="doc-param-default">
                    <b>DEFAULT:</b>
                      <code>None</code>
                  </span>
              </p>
            </td>
          </tr>
      </tbody>
    </table>


<table>
      <thead>
        <tr>
          <th><span class="doc-section-title">RETURNS</span></th>
          <th><span>DESCRIPTION</span></th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <span class="doc-returns-annotation">
                    <code><span title="pandas.DataFrame">DataFrame</span></code>
                </span>
            </td>
            <td class="doc-returns-details">
              <div class="doc-md-description">
                <p>An annotations dataframe with a single go term.</p>
              </div>
            </td>
          </tr>
      </tbody>
    </table>


    </div>

</div>

<div class="doc doc-object doc-function">


<h3 id="gopher.load_annotations" class="doc doc-heading">
            <code class="highlight language-python"><span class="n">gopher</span><span class="o">.</span><span class="n">load_annotations</span><span class="p">(</span><span class="n">species</span><span class="p">,</span> <span class="n">aspect</span><span class="o">=</span><span class="s1">&#39;all&#39;</span><span class="p">,</span> <span class="n">release</span><span class="o">=</span><span class="s1">&#39;current&#39;</span><span class="p">,</span> <span class="n">fetch</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span></code>

</h3>


    <div class="doc doc-contents first">

        <p>Load the Gene Ontology (GO) annotations for a species.</p>


<table>
      <thead>
        <tr>
          <th><span class="doc-section-title">PARAMETER</span></th>
          <th><span>DESCRIPTION</span></th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>species</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>The species for which to retrieve GO annotations. If not &ldquo;humnan&rdquo; or
&ldquo;yeast&rdquo;, see
<a href="http://current.geneontology.org/products/pages/downloads.html">here</a>.</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code>(<span title="str">str</span>, {<span title="human">human</span>, <span title="yeast">yeast</span>, ...})</code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>aspect</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>The Gene Ontology aspect to use. Use &ldquo;c&rdquo; for &ldquo;Cellular Compartment&rdquo;,
&ldquo;f&rdquo; for &ldquo;Molecular Function&rdquo;, or &ldquo;p&rdquo; for &ldquo;Biological Process&rdquo;. <code>None</code>
uses all of the them.</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code>(<span title="str">str</span>, {<span title="cc">cc</span>, <span title="mf">mf</span>, <span title="bp">bp</span>, <span title="all">all</span>})</code>
                  </span>
                  <span class="doc-param-default">
                    <b>DEFAULT:</b>
                      <code>&#39;all&#39;</code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>release</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>The Gene Ontology release version. Using &ldquo;current&rdquo; will look up the
most current version.</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="str">str</span></code>
                  </span>
                  <span class="doc-param-default">
                    <b>DEFAULT:</b>
                      <code>&#39;current&#39;</code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>fetch</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>Download the file even if it already exists?</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="bool">bool</span></code>
                  </span>
                  <span class="doc-param-default">
                    <b>DEFAULT:</b>
                      <code>False</code>
                  </span>
              </p>
            </td>
          </tr>
      </tbody>
    </table>


<table>
      <thead>
        <tr>
          <th><span class="doc-section-title">RETURNS</span></th>
          <th><span>DESCRIPTION</span></th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <span class="doc-returns-annotation">
                    <code><span title="pandas.DataFrame">DataFrame</span></code>
                </span>
            </td>
            <td class="doc-returns-details">
              <div class="doc-md-description">
                <p>The annotation dataframe.</p>
              </div>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <span class="doc-returns-annotation">
                    <code><span title="dict">dict</span></code>
                </span>
            </td>
            <td class="doc-returns-details">
              <div class="doc-md-description">
                <p>A mapping of GO terms (keys) to Uniprot accessions with that
annotation.</p>
              </div>
            </td>
          </tr>
      </tbody>
    </table>


    </div>

</div>

<div class="doc doc-object doc-function">


<h3 id="gopher.get_annotations" class="doc doc-heading">
            <code class="highlight language-python"><span class="n">gopher</span><span class="o">.</span><span class="n">get_annotations</span><span class="p">(</span><span class="n">proteins</span><span class="p">,</span> <span class="n">aspect</span><span class="o">=</span><span class="s1">&#39;all&#39;</span><span class="p">,</span> <span class="n">species</span><span class="o">=</span><span class="s1">&#39;human&#39;</span><span class="p">,</span> <span class="n">release</span><span class="o">=</span><span class="s1">&#39;current&#39;</span><span class="p">,</span> <span class="n">fetch</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">go_subset</span><span class="o">=</span><span class="kc">None</span><span class="p">)</span></code>

</h3>


    <div class="doc doc-contents first">

        <p>Gets the annotations for proteins in a dataset</p>


<table>
      <thead>
        <tr>
          <th><span class="doc-section-title">PARAMETER</span></th>
          <th><span>DESCRIPTION</span></th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>proteins</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>Dataframe of proteins and quantifications</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="pandas.DataFrame">DataFrame</span></code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>aspect</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>The Gene Ontology aspect to use. Use &ldquo;cc&rdquo; for &ldquo;Cellular Compartment&rdquo;,
&ldquo;mf&rdquo; for &ldquo;Molecular Function&rdquo;, &ldquo;bp&rdquo; for &ldquo;Biological Process&rdquo;, or &ldquo;all&rdquo;
for all three.</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code>(<span title="str">str</span>, {<span title="cc">cc</span>, <span title="mf">mf</span>, <span title="bp">bp</span>, <span title="all">all</span>})</code>
                  </span>
                  <span class="doc-param-default">
                    <b>DEFAULT:</b>
                      <code>&#39;all&#39;</code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>species</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>The species for which to retrieve GO annotations. If not &ldquo;human&rdquo; or
&ldquo;yeast&rdquo;, see
<a href="http://current.geneontology.org/products/pages/downloads.html">here</a>.</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code>str, {&#34;human&#34;, &#34;yeast&#34;, ...}, optional.</code>
                  </span>
                  <span class="doc-param-default">
                    <b>DEFAULT:</b>
                      <code>&#39;human&#39;</code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>release</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>The Gene Ontology release version. Using &ldquo;current&rdquo; will look up the
most current version.</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="str">str</span></code>
                  </span>
                  <span class="doc-param-default">
                    <b>DEFAULT:</b>
                      <code>&#39;current&#39;</code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>fetch</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>Download the GO annotations even if they have been downloaded before?</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="bool">bool</span></code>
                  </span>
                  <span class="doc-param-default">
                    <b>DEFAULT:</b>
                      <code>False</code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>go_subset</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>The go terms of interest. Should consists of the go term names such
as &lsquo;nucleus&rsquo; or &lsquo;cytoplasm&rsquo;.</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="list">list</span></code>
                  </span>
                  <span class="doc-param-default">
                    <b>DEFAULT:</b>
                      <code>None</code>
                  </span>
              </p>
            </td>
          </tr>
      </tbody>
    </table>


<table>
      <thead>
        <tr>
          <th><span class="doc-section-title">RETURNS</span></th>
          <th><span>DESCRIPTION</span></th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <span class="doc-returns-annotation">
                    <code><span title="pandas.DataFrame">DataFrame</span></code>
                </span>
            </td>
            <td class="doc-returns-details">
              <div class="doc-md-description">
                <p>Dataframe with protein annotations</p>
              </div>
            </td>
          </tr>
      </tbody>
    </table>


    </div>

</div>

<div class="doc doc-object doc-function">


<h3 id="gopher.map_proteins" class="doc doc-heading">
            <code class="highlight language-python"><span class="n">gopher</span><span class="o">.</span><span class="n">map_proteins</span><span class="p">(</span><span class="n">protein_list</span><span class="p">,</span> <span class="n">aspect</span><span class="o">=</span><span class="s1">&#39;all&#39;</span><span class="p">,</span> <span class="n">species</span><span class="o">=</span><span class="s1">&#39;human&#39;</span><span class="p">,</span> <span class="n">release</span><span class="o">=</span><span class="s1">&#39;current&#39;</span><span class="p">,</span> <span class="n">fetch</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span></code>

</h3>


    <div class="doc doc-contents first">

        <p>Map the proteins to the GO terms</p>


<table>
      <thead>
        <tr>
          <th><span class="doc-section-title">PARAMETER</span></th>
          <th><span>DESCRIPTION</span></th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>protein_list</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>A list of UniProt accessions.</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="List">List</span>[<span title="str">str</span>]</code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>aspect</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>The Gene Ontology aspect to use. Use &ldquo;cc&rdquo; for &ldquo;Cellular Compartment&rdquo;,
&ldquo;mf&rdquo; for &ldquo;Molecular Function&rdquo;, &ldquo;bp&rdquo; for &ldquo;Biological Process&rdquo;, or &ldquo;all&rdquo;
for all three.</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code>(<span title="str">str</span>, {<span title="cc">cc</span>, <span title="mf">mf</span>, <span title="bp">bp</span>, <span title="all">all</span>})</code>
                  </span>
                  <span class="doc-param-default">
                    <b>DEFAULT:</b>
                      <code>&#39;all&#39;</code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>species</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>The species for which to retrieve GO annotations. If not &ldquo;human&rdquo; or
&ldquo;yeast&rdquo;, see
<a href="http://current.geneontology.org/products/pages/downloads.html">here</a>.</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code>str, {&#34;human&#34;, &#34;yeast&#34;, ...}, optional.</code>
                  </span>
                  <span class="doc-param-default">
                    <b>DEFAULT:</b>
                      <code>&#39;human&#39;</code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>release</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>The Gene Ontology release version. Using &ldquo;current&rdquo; will look up the
most current version.</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="str">str</span></code>
                  </span>
                  <span class="doc-param-default">
                    <b>DEFAULT:</b>
                      <code>&#39;current&#39;</code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>fetch</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>Download the GO annotations even if they have been downloaded before?</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="bool">bool</span></code>
                  </span>
                  <span class="doc-param-default">
                    <b>DEFAULT:</b>
                      <code>False</code>
                  </span>
              </p>
            </td>
          </tr>
      </tbody>
    </table>


<table>
      <thead>
        <tr>
          <th><span class="doc-section-title">RETURNS</span></th>
          <th><span>DESCRIPTION</span></th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <span class="doc-returns-annotation">
                    <code><span title="pandas.DataFrame">DataFrame</span></code>
                </span>
            </td>
            <td class="doc-returns-details">
              <div class="doc-md-description">
                <p>Dataframe with protein accessions and GO terms</p>
              </div>
            </td>
          </tr>
      </tbody>
    </table>


    </div>

</div>

<div class="doc doc-object doc-function">


<h3 id="gopher.normalize_values" class="doc doc-heading">
            <code class="highlight language-python"><span class="n">gopher</span><span class="o">.</span><span class="n">normalize_values</span><span class="p">(</span><span class="n">proteins</span><span class="p">,</span> <span class="n">fasta</span><span class="p">)</span></code>

</h3>


    <div class="doc doc-contents first">

        <p>Normalize intensity values.</p>
<p>Normalize using the proteomic ruler approach outlined by Wiśniewski et al.
(doi: https://doi.org/10.1074/mcp.M113.037309)</p>


<table>
      <thead>
        <tr>
          <th><span class="doc-section-title">PARAMETER</span></th>
          <th><span>DESCRIPTION</span></th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>proteins</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>A dataframe where the indices are UniProt accessions and each column is
an experiment to test. The values in this dataframe raw protein abundance</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="pandas.DataFrame">DataFrame</span></code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>fasta</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>Use the FASTA file to generate molecular weights for normalization</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="pathlib.Path">Path</span></code>
                  </span>
              </p>
            </td>
          </tr>
      </tbody>
    </table>


<table>
      <thead>
        <tr>
          <th><span class="doc-section-title">RETURNS</span></th>
          <th><span>DESCRIPTION</span></th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <span class="doc-returns-annotation">
                    <code><span title="pandas.DataFrame">DataFrame</span></code>
                </span>
            </td>
            <td class="doc-returns-details">
              <div class="doc-md-description">
                <p>The normalized intensities for every protein in each sample.</p>
              </div>
            </td>
          </tr>
      </tbody>
    </table>


    </div>

</div>

<div class="doc doc-object doc-function">


<h3 id="gopher.get_rankings" class="doc doc-heading">
            <code class="highlight language-python"><span class="n">gopher</span><span class="o">.</span><span class="n">get_rankings</span><span class="p">(</span><span class="n">proteins</span><span class="p">,</span> <span class="n">go_term</span><span class="p">,</span> <span class="n">aspect</span><span class="o">=</span><span class="s1">&#39;all&#39;</span><span class="p">,</span> <span class="n">species</span><span class="o">=</span><span class="s1">&#39;human&#39;</span><span class="p">,</span> <span class="n">release</span><span class="o">=</span><span class="s1">&#39;current&#39;</span><span class="p">,</span> <span class="n">fetch</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span></code>

</h3>


    <div class="doc doc-contents first">

        <p>Rank the proteins and show whether proteins are in a specified term</p>


<table>
      <thead>
        <tr>
          <th><span class="doc-section-title">PARAMETER</span></th>
          <th><span>DESCRIPTION</span></th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>proteins</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>Dataframe of protein quant data</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="pandas.DataFrame">DataFrame</span></code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>go_term</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>String of specified GO term name</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="str">str</span></code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>aspect</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>The Gene Ontology aspect to use. Use &ldquo;cc&rdquo; for &ldquo;Cellular Compartment&rdquo;,
&ldquo;mf&rdquo; for &ldquo;Molecular Function&rdquo;, &ldquo;bp&rdquo; for &ldquo;Biological Process&rdquo;, or &ldquo;all&rdquo;
for all three.</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code>(<span title="str">str</span>, {<span title="cc">cc</span>, <span title="mf">mf</span>, <span title="bp">bp</span>, <span title="all">all</span>})</code>
                  </span>
                  <span class="doc-param-default">
                    <b>DEFAULT:</b>
                      <code>&#39;all&#39;</code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>species</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>The species for which to retrieve GO annotations. If not &ldquo;human&rdquo; or
&ldquo;yeast&rdquo;, see
<a href="http://current.geneontology.org/products/pages/downloads.html">here</a>.</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code>str, {&#34;human&#34;, &#34;yeast&#34;, ...}, optional.</code>
                  </span>
                  <span class="doc-param-default">
                    <b>DEFAULT:</b>
                      <code>&#39;human&#39;</code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>release</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>The Gene Ontology release version. Using &ldquo;current&rdquo; will look up the
most current version.</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="str">str</span></code>
                  </span>
                  <span class="doc-param-default">
                    <b>DEFAULT:</b>
                      <code>&#39;current&#39;</code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>fetch</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>Download the GO annotations even if they have been downloaded before?</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="bool">bool</span></code>
                  </span>
                  <span class="doc-param-default">
                    <b>DEFAULT:</b>
                      <code>False</code>
                  </span>
              </p>
            </td>
          </tr>
      </tbody>
    </table>


<table>
      <thead>
        <tr>
          <th><span class="doc-section-title">RETURNS</span></th>
          <th><span>DESCRIPTION</span></th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <span class="doc-returns-annotation">
                    <code><span title="pandas.DataFrame">DataFrame</span></code>
                </span>
            </td>
            <td class="doc-returns-details">
              <div class="doc-md-description">
                <p>Dataframe with protein rankings and whether or not the protein is in the specified term</p>
              </div>
            </td>
          </tr>
      </tbody>
    </table>


    </div>

</div>

<div class="doc doc-object doc-function">


<h3 id="gopher.in_term" class="doc doc-heading">
            <code class="highlight language-python"><span class="n">gopher</span><span class="o">.</span><span class="n">in_term</span><span class="p">(</span><span class="n">proteins</span><span class="p">,</span> <span class="n">go_term</span><span class="p">,</span> <span class="n">annot</span><span class="p">)</span></code>

</h3>


    <div class="doc doc-contents first">

        <p>See if proteins are associated with a specific term</p>


<table>
      <thead>
        <tr>
          <th><span class="doc-section-title">PARAMETER</span></th>
          <th><span>DESCRIPTION</span></th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>proteins</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>Dataframe of proteins and quantifications</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="pandas.DataFrame">DataFrame</span></code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>go_term</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>String of specified GO term name</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="str">str</span></code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>annot</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>Annotation file for the dataset</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="pandas.DataFrame">DataFrame</span></code>
                  </span>
              </p>
            </td>
          </tr>
      </tbody>
    </table>


<table>
      <thead>
        <tr>
          <th><span class="doc-section-title">RETURNS</span></th>
          <th><span>DESCRIPTION</span></th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <span class="doc-returns-annotation">
                    <code><span title="pandas.DataFrame">DataFrame</span></code>
                </span>
            </td>
            <td class="doc-returns-details">
              <div class="doc-md-description">
                <p>Dataframe with protein quant and if protein is in the given term</p>
              </div>
            </td>
          </tr>
      </tbody>
    </table>


    </div>

</div>

<div class="doc doc-object doc-function">


<h3 id="gopher.roc" class="doc doc-heading">
            <code class="highlight language-python"><span class="n">gopher</span><span class="o">.</span><span class="n">roc</span><span class="p">(</span><span class="n">proteins</span><span class="p">,</span> <span class="n">go_term</span><span class="p">,</span> <span class="n">aspect</span><span class="o">=</span><span class="s1">&#39;all&#39;</span><span class="p">,</span> <span class="n">species</span><span class="o">=</span><span class="s1">&#39;human&#39;</span><span class="p">,</span> <span class="n">release</span><span class="o">=</span><span class="s1">&#39;current&#39;</span><span class="p">,</span> <span class="n">fetch</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span></code>

</h3>


    <div class="doc doc-contents first">

        <p>Plot the ROC curve for a go term in each sample</p>


<table>
      <thead>
        <tr>
          <th><span class="doc-section-title">PARAMETER</span></th>
          <th><span>DESCRIPTION</span></th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>proteins</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>Dataframe of proteins and quantifications</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="pandas.DataFrame">DataFrame</span></code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>go_term</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>String of specified GO term name</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="str">str</span></code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>aspect</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>The Gene Ontology aspect to use. Use &ldquo;cc&rdquo; for &ldquo;Cellular Compartment&rdquo;,
&ldquo;mf&rdquo; for &ldquo;Molecular Function&rdquo;, &ldquo;bp&rdquo; for &ldquo;Biological Process&rdquo;, or &ldquo;all&rdquo;
for all three.</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code>(<span title="str">str</span>, {<span title="cc">cc</span>, <span title="mf">mf</span>, <span title="bp">bp</span>, <span title="all">all</span>})</code>
                  </span>
                  <span class="doc-param-default">
                    <b>DEFAULT:</b>
                      <code>&#39;all&#39;</code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>species</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>The species for which to retrieve GO annotations. If not &ldquo;human&rdquo; or
&ldquo;yeast&rdquo;, see
<a href="http://current.geneontology.org/products/pages/downloads.html">here</a>.</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code>str, {&#34;human&#34;, &#34;yeast&#34;, ...}, optional.</code>
                  </span>
                  <span class="doc-param-default">
                    <b>DEFAULT:</b>
                      <code>&#39;human&#39;</code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>release</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>The Gene Ontology release version. Using &ldquo;current&rdquo; will look up the
most current version.</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="str">str</span></code>
                  </span>
                  <span class="doc-param-default">
                    <b>DEFAULT:</b>
                      <code>&#39;current&#39;</code>
                  </span>
              </p>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>fetch</code>
            </td>
            <td class="doc-param-details">
              <div class="doc-md-description">
                <p>Download the GO annotations even if they have been downloaded before?</p>
              </div>
              <p>
                  <span class="doc-param-annotation">
                    <b>TYPE:</b>
                      <code><span title="bool">bool</span></code>
                  </span>
                  <span class="doc-param-default">
                    <b>DEFAULT:</b>
                      <code>False</code>
                  </span>
              </p>
            </td>
          </tr>
      </tbody>
    </table>


<table>
      <thead>
        <tr>
          <th><span class="doc-section-title">RETURNS</span></th>
          <th><span>DESCRIPTION</span></th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <span class="doc-returns-annotation">
                    <code><span title="matplotlib.pyplot">pyplot</span></code>
                </span>
            </td>
            <td class="doc-returns-details">
              <div class="doc-md-description">
                <p>Plot of ROC curve for a GO term</p>
              </div>
            </td>
          </tr>
      </tbody>
    </table>


    </div>

</div>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
        <div class="md-social">
  
    
    
    
    
      
      
    
    <a href="https://twitter.com/talusbio" target="_blank" rel="noopener" title="twitter.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.7.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253"/></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "..", "features": ["toc.integrate", "navigation.tabs"], "search": "../assets/javascripts/workers/search.f8cc74c7.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../assets/javascripts/bundle.c8b220af.min.js"></script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
      
    
  </body>
</html>